wgsl/index.bs

```diff
@@ -3406,11 +3815,95 @@ Issue: Which index is used when it's out of bounds?
 
 TODO: *Stub*. Call to function that has a [=return type=] is an expression.
 
-## Variable or const reference TODO ## {#var-const-ref-expr}
+## Variable Identifier Expression ## {#var-identifier-expr}
+
+<table class='data'>
+  <caption>Getting a reference from a variable name</caption>
+  <thead>
+    <tr><th>Precondition<th>Conclusion<th>Description
+  </thead>
+  <tr algorithm="variable reference">
+       <td>
+          |v| is an identifier [=resolves|resolving=] to
+          an [=in scope|in-scope=] variable declared in [=storage class=] |SC|
+          with [=store type=] |T|
+       <td class="nowrap">
+          |v| : ref&lt;|SC|,|T|&gt;
+       <td>Result is a reference to the storage for the named variable |v|.
+</table>
+
+## Const Identifier Expression  ## {#const-identifier-expr}
```

Comment:
this seems unrelated to the pointers

Reply:
I did it now because it went along with giving the rule for variables, but the variable one is tied to the references concept.

Arguably, it is tied to this PR because it's precise about the fact the identifier denotes the *evaluation* of the value at the time of initialization, and built into that is the Load Rule.  Arguably it can be landed independently.  Since it appears you think this can land independently, I'll pull it out.

Reply:
I've pulled this out to #1586